core/src/test/java/io/questdb/griffin/TableWriterAsyncCmdTest.java

```diff
@@ -0,0 +1,346 @@
+        assertMemoryLeak(() -> {
+            compile("create table product (timestamp timestamp)", sqlExecutionContext);
+
+            // Block event queue with stale sequence
+            SCSequence staleSequence = new SCSequence();
+            setUpEngineAsyncWriterEventWait(engine, staleSequence);
+
+            try (TableWriter writer = engine.getWriter(sqlExecutionContext.getCairoSecurityContext(), "product", "test lock")) {
+                for (int i = 0; i < engineEventQueue; i++) {
+                    CompiledQuery cc = compiler.compile("ALTER TABLE product add column column" + i + " int", sqlExecutionContext);
+                    executeAlterCommandNoWait(engine, cc.getAlterStatement(), sqlExecutionContext);
+                    engine.tick();
+                    writer.tick();
+                }
+
+                // Add column when event queue is stalled
+                CompiledQuery cc = compiler.compile("ALTER TABLE product add column column5 int", sqlExecutionContext);
+                setUpEngineAsyncWriterEventWait(engine, commandReplySequence);
+                long commandId = executeAlterCommandNoWait(engine, cc.getAlterStatement(), sqlExecutionContext);
+                engine.tick();
+                writer.tick();
+                SqlException exception = waitWriterEvent(engine, commandId, commandReplySequence, 500_000, 0);
+                TestUtils.assertContains(exception.getFlyweightMessage(), "Timeout expired");
+
+                // Remove sequence
+                stopEngineAsyncWriterEventWait(engine, staleSequence);
+
+                // Re-execute last query
+                try {
+                    setUpEngineAsyncWriterEventWait(engine, commandReplySequence);
+                    commandId = executeAlterCommandNoWait(engine, cc.getAlterStatement(), sqlExecutionContext);
+                    engine.tick();
+                    writer.tick();
+
+                    exception = waitWriterEvent(engine, commandId, commandReplySequence, 500_000, 0);
+                    TestUtils.assertContains(exception.getFlyweightMessage(), "Duplicate column name: column5");
+                } finally {
+                    stopEngineAsyncWriterEventWait(engine, commandReplySequence);
+                }
+            }
+        });
+    }
+
+    @Test
+    public void testAsyncAlterCommandsExceedsEngineCmdQueue() throws Exception {
+        assertMemoryLeak(() -> {
+            compile("create table product (timestamp timestamp)", sqlExecutionContext);
+
+            // Block table
+            try (TableWriter ignored = engine.getWriter(sqlExecutionContext.getCairoSecurityContext(), "product", "test lock")) {
+
+                for (int i = 0; i < engineCmdQueue; i++) {
+                    CompiledQuery cc = compiler.compile("ALTER TABLE product add column column" + i + " int", sqlExecutionContext);
+                    executeAlterCommandNoWait(engine, cc.getAlterStatement(), sqlExecutionContext);
```

Comment:
move tests on the new API and remove `getAlterStatement()` method?

Reply:
done